### DISCUSSION TITLE
Streaming 10 rows at time

### DISCUSSION THREAD

**Discussion by ericsouza** [sqlalchemy/sqlalchemy#7298] [Usage Questions] [answered]
Hi, I'm using sqlalchemy 1.4+ and I need to get results from a table and send a message to sqs using batch, which receives 10 messages at time. So, I wonder if its possible to have some similar to the following code:

```python
engine: AsyncEngine = create_async_engine(
    "mysql+asyncmy://user:password@localhost:3306/test",
    echo=False
)

query = "some query"
async def main():
    async with engine.connect() as conn:
            result = await conn.stream(query)
            async for rows in result:
                print(len(rows))  # should print 10
                await send_batch_messages(rows)

asyncio.run(main())
```

**Comment by zzzeek** [comment, marked as answer]
we have exactly the API for this, take a look at [partitions](https://docs.sqlalchemy.org/en/14/orm/extensions/asyncio.html?highlight=asyncresult%20partitions#sqlalchemy.ext.asyncio.AsyncResult.partitions)

**Reply by ericsouza** [reply]
Oh, it seems I didn't look carefully at the docs. Thanks 😄
